File: backend/src/main/java/org/camunda/optimize/dto/optimize/query/report/single/configuration/UserTaskDurationTime.java
/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */
package org.camunda.optimize.dto.optimize.query.report.single.configuration;

import com.fasterxml.jackson.annotation.JsonValue;

import static org.camunda.optimize.dto.optimize.ReportConstants.IDLE_USER_TASK_DURATION_TIME;
import static org.camunda.optimize.dto.optimize.ReportConstants.TOTAL_USER_TASK_DURATION_TIME;
import static org.camunda.optimize.dto.optimize.ReportConstants.WORK_USER_TASK_DURATION_TIME;

public enum UserTaskDurationTime {
  IDLE(IDLE_USER_TASK_DURATION_TIME),
  WORK(WORK_USER_TASK_DURATION_TIME),
  TOTAL(TOTAL_USER_TASK_DURATION_TIME),
  ;

  private final String id;

  UserTaskDurationTime(final String id) {
    this.id = id;
  }

  @JsonValue
  public String getId() {
    return id;
  }

  @Override
  public String toString() {
    return getId();
  }
}

File: client/src/components/Reports/controlPanels/Configuration/UserTaskDurationTime.js
/*
 * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
 * under one or more contributor license agreements. Licensed under a commercial license.
 * You may not use this file except in compliance with the commercial license.
 */

import React from 'react';

import {Select} from 'components';
import {t} from 'translation';

export default function UserTaskDurationTime({
  report: {
    data: {configuration, view}
  },
  onChange
}) {
  if (view && view.entity === 'userTask' && view.property === 'duration') {
    return (
      <fieldset className="UserTaskDurationTime">
        <legend>{t('report.config.userTaskDuration.legend')}</legend>
        <Select
          value={configuration.userTaskDurationTime}
          onChange={value => onChange({userTaskDurationTime: {$set: value}}, true)}
        >
          <Select.Option value="idle">{t('report.config.userTaskDuration.idle')}</Select.Option>
          <Select.Option value="work">{t('report.config.userTaskDuration.work')}</Select.Option>
          <Select.Option value="total">{t('report.config.userTaskDuration.total')}</Select.Option>
        </Select>
      </fieldset>
    );
  }
  return null;
}
